client/transaction-pool/graph/src/pool.rs

```diff
@@ -438,6 +447,16 @@ impl<B: ChainApi> Pool<B> {
 		(hash, validity)
 	}
 
+	/// Notify all watchers that transactions in the block with hash been finalized
+	pub async fn finalized(&self, block_hash: BlockHash<B>) -> Result<(), B::Error> {
```

Comment:
There is already a method called `on_broadcasted`, I think it would be good to keep the convention and call this one `on_block_finalized`
```suggestion
	pub async fn on_block_finalized(&self, block_hash: BlockHash<B>) -> Result<(), B::Error> {
```

Basically to separate functions that control internal state of the pool from functions that are just handlers for some external events.